content/chainguard/chainguard-enforce/authentication/identity-examples/enforce-aws-identity.md

```diff
@@ -0,0 +1,359 @@
+    aws_user_id_pattern = "^AROA(.*):[AWS-ROLE-SESSION-NAME]$"
+
+    // NB: This role will be assumed so can't use the role ARN directly. We must used the ARN of the assumed role
+    aws_arn = "arn:aws:sts::[AWS-ACCOUNT-ID]:assumed-role/[AWS-ROLE-NAME]/[AWS-ROLE-SESSION-NAME]"
+  }
+}
+```
+
+First this section creates a Chainguard Identity tied to the `chainguard_group` created by the `sample.tf` file; namely, the `example-group` group. The identity is named `aws-identity` and has a brief description.
+
+The most important part of this section is the `aws_identity`. When the AWS role tries to assume this identity later on, it must present a token matching the `aws_account`, `aws_user_id_pattern`, and `aws_arn` specified here in order to do so.
+
+The next section will output the new identity's `id` value. This is a unique value that represents the identity itself.
+
+```
+output "aws-identity" {
+  value = chainguard_identity.aws.id
+}
+```
+
+The section after that looks up the `viewer` role.
+
+```
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+```
+
+The final section grants this role to the identity on the `example-group`.
+
+```
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.aws.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+```
+
+After defining these resources, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+## Using the identity with AWS Lambda
+
+In this section, you'll create an AWS Lambda function that will assume the identity you created in the previous section. This function will then use the identity to interact with Chainguard resources.
+
+First, define the following resources in a `lambda.tf` file. The file will consist of three sections, which we'll go over one by one.
+
+This section defines an IAM policy document that will allow the AWS Lambda service to assume the role you'll create in the next section:
+
+```hcl
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+```
+
+This section defines the role that the AWS Lambda service will assume:
+
+```hcl
+resource "aws_iam_role" "iam_for_lambda" {
+  name               = "iam_for_lambda"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+```
+
+This section defines a AWS lambda function implemented in Node.js that will assume the identity you created in the previous section:
+
+```hcl
+resource "aws_lambda_function" "test_lambda" {
+  filename      = "lambda_function_payload.zip"
+  function_name = "lambda_function_name"
+  role          = aws_iam_role.iam_for_lambda.arn
+  handler       = "index.test"
+
+  source_code_hash = data.archive_file.lambda.output_base64sha256
```

Comment:
I trimmed out a bit of stuff for brevity, but I can link to where it's described more in detail in TF and AWS docs.

I wanted to avoid making this a "how to deploy a Go Lambda function" doc, since many readers may not be doing that, it's just necessary to test the identity.